src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc

```diff
@@ -0,0 +1,218 @@
+  LOG4CXX_AUTO_TRACE(logger_);
+  auto subscribe_waypoints_msg =
+      application_manager::MessageHelper::CreateMessageForHMI(
+          hmi_apis::FunctionID::Navigation_SubscribeWayPoints,
+          application_manager_.GetNextHMICorrelationID());
+  (*subscribe_waypoints_msg)[application_manager::strings::params]
+                            [application_manager::strings::message_type] =
+                                hmi_apis::messageType::request;
+  return subscribe_waypoints_msg;
+}
+
+void SDLPendingResumptionHandler::OnResumptionRevert() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  using namespace application_manager;
+
+  if (!pending_requests_.empty()) {
+    LOG4CXX_DEBUG(logger_, "Still waiting for some response");
+    return;
+  }
+
+  if (!freezed_resumptions_.empty()) {
+    ResumptionAwaitingHandling freezed_resumption = freezed_resumptions_.back();
+    freezed_resumptions_.pop_back();
+
+    auto request = std::make_shared<smart_objects::SmartObject>(
+        freezed_resumption.request_to_send_.message);
+    const uint32_t cid =
+        (*request)[strings::params][strings::correlation_id].asUInt();
+    const auto fid = static_cast<hmi_apis::FunctionID::eType>(
+        (*request)[strings::params][strings::function_id].asInt());
+
+    LOG4CXX_DEBUG(logger_,
+                  "Subscribing for event with function id: "
+                      << fid << " correlation id: " << cid);
+    subscribe_on_event(fid, cid);
+    pending_requests_[cid] = *request;
+    LOG4CXX_DEBUG(logger_,
+                  "Sending request with fid: " << fid << " and cid: " << cid);
+    application_manager_.GetRPCService().ManageHMICommand(request);
+  }
+}
+
+void SDLPendingResumptionHandler::RaiseFakeSuccessfulResponse(
+    ns_smart_device_link::ns_smart_objects::SmartObject response,
+    const int32_t corr_id) {
+  using namespace application_manager;
+  response[strings::params][strings::correlation_id] = corr_id;
+  auto fid = static_cast<hmi_apis::FunctionID::eType>(
+      response[strings::params][strings::function_id].asInt());
+  event_engine::Event event(fid);
+  event.set_smart_object(response);
+
+  LOG4CXX_TRACE(logger_,
+                "Raise fake response for subscriber. corr_id : " << corr_id);
+  event.raise(application_manager_.event_dispatcher());
+}
+
+void SDLPendingResumptionHandler::on_event(
+    const application_manager::event_engine::Event& event) {
+  using namespace application_manager;
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const smart_objects::SmartObject& response = event.smart_object();
+  const uint32_t corr_id = event.smart_object_correlation_id();
+
+  LOG4CXX_TRACE(logger_,
+                "Received event with function id: "
+                    << event.id() << " and correlation id: " << corr_id);
+
+  smart_objects::SmartObject pending_request;
+  if (pending_requests_.find(corr_id) == pending_requests_.end()) {
+    LOG4CXX_DEBUG(logger_, "corr id " << corr_id << " NOT found");
```

Comment:
ERROR

Reply:
Done in 25b2f2299d5caa7432c0e51f6208f3696143dc85